cgi-bin/Apache/LiveJournal.pm

```diff
@@ -1129,6 +1182,31 @@ sub vgift_content {
     return OK;
 }
 
+# load_file_for_concat will retrieve the given file from the disk and return:
+# (contents, size, mtime, mime). If the file doesn't exist, undef is returned.
+sub load_file_for_concat {
+    my $fn = $_[0];
+    my @stat = stat( $fn );
+    return undef unless scalar @stat > 0;
+    return undef unless -f $fn;
```

Comment:
I think the info used by -f is already in $stat[2] if you want to avoid statting it twice, but it will probably not look elegant.